### ISSUE TITLE
Invalid dub.sdl should fail on build, not just describe

### ISSUE THREAD

**Issue by rcorre** [D-Programming-Language/dub#711]
I had the following in my `dub.sdl`, which I _thought_ was a way to customize the `unittest` build while keeping `library` as the default (by making it the first config):

```
configuration "library"

configuration "unittest" {
  importPaths "src" "tests"
  sourcePaths "src" "tests"
}
```

It turns out this is incorrect syntax (Configurations must have a concrete target type.), but it took me awhile to figure this out, as it would build without complaining. It only failed when I ran `dub describe`.
It seems like either:

1. This is incorrect, in which case _both_ `describe` and `build` should fail
2. This is correct, in which case `describe` should succeed.

If my dub.sdl is incorrect, I'd rather know sooner than later.

**Comment by s-ludwig** [comment]
Hm, this actually seems to be a bug in the SDL parser code. Configurations should inherit to the global `targeType` and if that isn't set, or set to `autodetect`, default to `library`.

**Comment by rcorre** [comment]
So is the first `configuration` line (with no following block) a valid way to specify the default configuration?

**Comment by s-ludwig** [comment]
Yes, that should otherwise work.

**Comment by s-ludwig** [comment]
Fixed by 4a8caa2c44be4a4c711b57cea008707453de1619.
